compiler/p/runtime/PPCCodeSync.inc

```diff
@@ -112,7 +112,7 @@ void ppcCodeSync(unsigned char *codeStart, unsigned int codeSize)
 	for(addr = codeStart ; addr < limit; addr += cacheLineSize)
 	   {
 
-		#if defined(__IBMC__) || defined(__IBMCPP__)
+		#if defined(__IBMC__) || defined(__IBMCPP__) && !defined(LINUX)
 		icbi(addr);
 		#elif defined(LINUX)
 		__asm__(
```

Comment:
I think we can simplify this by defining a suitable `icbi` at the top of this file where similar macros are defined. Can you take a crack at defining everything at the top and using it in the rest of the code and cleaning up this mix and match we have going on here? In general prefer to use builtins when available, then inline assembly for GCC and new XLC, and either inline assembly or `mc_funcs` for old XLC, though I would lean towards inline assembly.